### ISSUE TITLE
[java] DoubleCheckedLocking: False negative when the outer null check is written as !(x != null)

### ISSUE THREAD

**Issue by MarkLee131** [pmd/pmd#6900]
**Affects PMD Version:** 7.26.0

**Rule:** DoubleCheckedLocking
<https://docs.pmd-code.org/latest/pmd_rules_java_multithreading.html#doublecheckedlocking>

**Description:**

`DoubleCheckedLocking` matches the guard by its syntactic form. Writing the outer null check as the logically equivalent `!(x != null)` instead of `x == null` makes the rule stop firing, even though the double-checked-locking anti-pattern (and its unsafe publication) is entirely unchanged. Found by metamorphic testing; the same silence reproduces across many double-checked-locking seeds under equivalent guard rewrites.

**Code Sample demonstrating the issue:**

```java
public class DclSeed {          // reported: DclSeed.java:4 DoubleCheckedLocking
    Object baz;

    Object bar() {
        if (baz == null) {
            synchronized (this) {
                if (baz == null) {
                    baz = new Object();
                }
            }
        }
        return baz;
    }
}
```

```java
public class DclVariant {       // NOT reported (only LogicInversion fires)
    Object baz;

    Object bar() {
        if (!(baz != null)) {                 // <- equivalent rewrite of `baz == null`
            synchronized (this) {
                if (baz == null) {
                    baz = new Object();
                }
            }
        }
        return baz;
    }
}
```

```
$ pmd check -d DclSeed.java,DclVariant.java \
      -R category/java/multithreading.xml/DoubleCheckedLocking -f text
DclSeed.java:4:  DoubleCheckedLocking: Double checked locking is not thread safe in Java.
# (no violation on DclVariant.java)
```

**Expected outcome:**

PMD should report a `DoubleCheckedLocking` violation at `DclVariant.java:4` (the outer `if`), but doesn't. This is a false-negative.

**Running PMD through:** CLI
